### ISSUE TITLE
State change causes all NestedRows to collapse

### ISSUE THREAD

**Issue by stokesbga** [fjmorant/react-native-nested-listview#81]
Is there a way to prevent all nested rows to close on press calling setState?

I have a list that I want to allow a press that changes the style of the nested row without closing it

**Comment by fjmorant** [comment]
Hi @stokesbga ,

Thank you very much for using this library, let me put an example : 

```
<NestedListView
  data={data}
  getChildrenName={(node) => 'items'}
  onNodePressed={(node) => alert('Selected node')}
  renderNode={(node, level) => (
    <NestedRow
      level={level}
      style={styles.row}
    >
      <Text>{node.title}</Text>
    </NestedRow>
  )}
/>
```

if you you change the data prop, it will refresh the NestedListView because there is new data to display, is this your case? 

THanks

**Comment by stokesbga** [comment]
I actually am not changing the data prop. it's a static array, but my state does change. I want a re render, but for the list to remain open/closed. state only manages the selected option in a static list of data.

I looked at nodeview, and noticed onpressnode called setstate ->opened: !opened. it would be nice to have the callback passed to the onpress prop to prevent that change, but still do a re render

**Comment by fjmorant** [comment]
Hi @stokesbga ,

could you post an example?

Thx

**Comment by stokesbga** [comment]
`class Hello extends Component {
constructor(props) {
  super(props);

  this.state = {
    category: {}
  }
}

render() {
  return (
        <NestedListView
            data={_.cloneDeep(this.props.allOptions.categories)}
            getChildrenName={(i) => 'children'}
            onNodePressed={(cat) => this._onPressCategory(cat)}
            renderNode={(item, level) => (
              <NestedRow level={level} style={{marginLeft: -10}}>
                <View style={[styles.touchableRowView, { backgroundColor: (this.state.category.id === item.id || this.state.category.parentId === item.id) ? '#eee' : '#fafafa' }]}>
                  <CheckBox checked={(this.state.category.id === item.id || this.state.category.parentId === item.id)} style={styles.checkbox}/>
                  <Text style={styles.filterLabel}>{item.name}</Text>
                  <Icon style={{color: `transparent`}} name={`ios-home`} />
                </View>
              </NestedRow>
  )
}
}`

**Comment by fjmorant** [comment]
Hi @stokesbga ,

I managed to change one of the examples to change the state in the wrapper component like your example : 

```
/* @flow */

import React from 'react'
import {
  Alert,
  StyleSheet,
  Text,
  View,
  type Props,
  type State,
} from 'react-native'
import NestedListView, {NestedRow} from 'react-native-nested-listview'

const generateXNumItems = (numItems, prefix) => {
  const items = []

  let i

  for (i = 0; i < numItems; i++) {
    items.push({
      name: `${prefix}.${i}`,
    })
  }

  return items
}

const data = [
  {
    name: 'Item level 1.1',
    descendants: generateXNumItems(10, 'Item level 1.1'),
  },
  {
    name: 'Item level 1.2',
    descendants: [
      {
        name: 'Item level 1.2.1',
      },
      {
        name: 'Item level 1.2.2',
        descendants: generateXNumItems(25, 'Item level 1.2.2'),
      },
    ],
  },
  {
    name: 'Item level 1.3',
    descendants: generateXNumItems(50, 'Item level 1.3'),
  },
]

const colorLevels = {
  [0]: 'white',
  [1]: 'blue',
  [2]: 'green',
  [3]: 'red',
}

const styles = StyleSheet.create({
  container: {flex: 1, backgroundColor: 'rgb(255, 255, 255)', padding: 15},
  node: {
    flex: 1,
    padding: 10,
    borderWidth: 1,
    borderColor: 'rgb(0, 0, 0)',
  },
})
export default class ExampleApp extends React.Component<Props, State> {
  nestedListView: any

  constructor(props) {
    super(props)

    this.state = {
      category: {},
    }
  }

  render() {
    alert('render example')
    return (
      <NestedListView
        data={data}
        getChildrenName={() => 'descendants'}
        onNodePressed={category => {
          this.setState({category})
        }}
        renderNode={(item, level) => (
          <NestedRow level={level} style={{marginLeft: -10}}>
            <View
              style={[
                styles.node,
                {
                  backgroundColor: colorLevels[level] || 'white',
                  paddingLeft: (level + 1) * 30,
                },
              ]}>
              <Text>{item.name}</Text>
            </View>
          </NestedRow>
        )}
      />
    )
  }
}

```

And the render method of the example gets called (obviously because the state changes) but the NestedListView component stays the same, the only thing that changes is the node you tapped.

I will continue investigating this issue, are you using the latest version (0.5.0)?

Thanks

**Comment by fjmorant** [comment]
Already solved in 0.6.0
